### PR TITLE
Increase statistics test coverage

### PR DESCRIPTION
**Pull request by magalilemes**
This PR is part of #276. Test coverage was increased to ~93%, according to [Codecov](https://codecov.io/gh/magalilemes/kworkflow/src/increase_statistics_test/src/statistics.sh).

### CODE REVIEW

File: tests/statistics_test.sh [hunk 1]
```diff
@@ -67,6 +72,22 @@ function statistics_Test
   output=$(statistics --invalid-option)
   msg="Invalid parameter: --invalid-option"
   assertEquals "($LINENO)" "$msg" "$output"
+
+  output=$(statistics --day not_a_day 2> /dev/null)
```
Comment (rodrigosiqueira) [inline tests/statistics_test.sh:76]: Double-quotes

File: tests/statistics_test.sh [hunk 2]
```diff
@@ -67,6 +72,22 @@ function statistics_Test
...
+  msg="Invalid parameter: not_a_day"
```
Comment (rodrigosiqueira) [inline tests/statistics_test.sh:77]: Single-quote

File: tests/statistics_test.sh [hunk 3]
```diff
@@ -67,6 +72,22 @@ function statistics_Test
...
+  assertEquals "($LINENO)" "$msg" "$output"
+
+  output=$(statistics --month not_a_month 2> /dev/null)
```
Comment (rodrigosiqueira) [inline tests/statistics_test.sh:80]: Fwiu here, you're testing an empty value which is nice. However, how about adding another test with an invalid month?

File: tests/statistics_test.sh [hunk 4]
```diff
@@ -67,6 +72,22 @@ function statistics_Test
...
+  msg="Invalid parameter: not_a_month"
+  assertEquals "($LINENO)" "$msg" "$output"
+
+  output=$(statistics --year not_a_year 2> /dev/null)
```
Comment (rodrigosiqueira) [inline tests/statistics_test.sh:84]: Try an invalid year.

File: tests/statistics_test.sh [hunk 5]
```diff
@@ -159,27 +179,43 @@ function day_statistics_Test
 
   day_data=$(day_statistics "$base_statistics/05/28")
   assertEquals "($LINENO)" "$msg2" "$day_data"
+
+  day_data=$(day_statistics "$base_statistics/05/27" | tail -n 2)
```
Comment (rodrigosiqueira) [inline tests/statistics_test.sh:183]: Nice use of `tail` :)

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]
Hi @magalilemes ,

First of all, thanks a lot for your patch. It is always good to see more tests :)

I added a few inline comments; please take a look at them. Also, review the single- vs. double-quote usage in your patch.

File: tests/statistics_test.sh (hunk 1 above)
Comment (magalilemes) [reply tests/statistics_test.sh:76]: Where in `output=$(statistics --day not_a_day 2> /dev/null)` exactly?

File: tests/statistics_test.sh (hunk 3 above)
Comment (magalilemes) [reply tests/statistics_test.sh:80]: not_a_month isn't an empty value. I meant it to be an invalid month, somehow. Like someone was trying to invoke `kw statistics --month not_a_month`.

File: tests/statistics_test.sh (hunk 1 above)
Comment (rodrigosiqueira) [reply tests/statistics_test.sh:76]: My bad... ignore my comment here.

File: tests/statistics_test.sh (hunk 3 above)
Comment (rodrigosiqueira) [reply tests/statistics_test.sh:80]: Hmm... correct if I'm wrong, but `not_a_month` and `not_a_year` are just a random string that you're passing to `statistics`.  The test that I suggested in my previous comment was using something like `13` or `-3` in the month parameter.

File: tests/statistics_test.sh (hunk 1 above)
Comment (rodrigosiqueira) [reply tests/statistics_test.sh:76]: Ignore my previous comment.

**Comment by rodrigosiqueira** [comment]
@magalilemes rebase your code.

**Comment by rodrigosiqueira** [comment]
Thanks a lot for this patch!
Merged, see: https://github.com/kworkflow/kworkflow/commit/f98c639e44da47a8ae8bc5bdd36ce569c0fc59df

**Pull request closed without merging**